test/mssql_connector/wait_for_mssql

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -ex
+
+function main {
+    wait_for_mssql
+}
+
+# FROM cyberark/bash-lib
+# Retry a command multiple times until it succeeds, with escalating
+# delay between attempts.
+# Delay is 2 * n + random up to 30s, then 30s + random after that.
+# For large numbers of retries the max delay is effectively the retry
+# in minutes.
+# Based on:
+# https://gist.github.com/sj26/88e1c6584397bb7c13bd11108a579746
+# but now quite heavily modified.
+function retry {
+    # Maxiumum amount of fixed delay between attempts
+    # a random value will still be added.
+    local -r MAX_BACKOFF=30
+
+    if [[ ${#} -lt 2 ]]; then
+        echo "retry usage: retry <retries> <command>"
+        exit 1
+    fi
+
+    local retries=$1
+    shift
+
+    if ! [[ ${retries} =~ ^[0-9\.]*$ ]]; then
+        echo "Invalid number of retries: ${retries} for command '${*}'".
+        exit 1
+    fi
+
+    local count=0
+    until eval "$@"; do
+        # Command failed, otherwise until would have skipped the loop
+
+        # Store return code so it can be reported to the user
+        exit=$?
+        count=$((count + 1))
+        if [ "${count}" -lt "${retries}" ]; then
+            # There are still retries left, calculate delay and notify user.
+            backoff=$((2 * count))
+            if [[ "${backoff}" -gt "${MAX_BACKOFF}" ]]; then
+                backoff=${MAX_BACKOFF}
+            fi;
+
+            # Add a random amount to the delay to prevent competing processes
+            # from re-colliding.
```

Comment:
This comment isn't clear to me, whatever the reason is that we're doing this.... since the delay is random, it could only make what we're trying to avoid slightly less likely.  It can't prevent it.

However, that is a kind of a moot point for me because I have a larger issue:

Imo using exponential backoff here is unnecessary complexity for little to no benefit.  I think a simpler strategy with less maintenance burden would just be:

1. Decide on a max time we give up after.  Say, 30s.
2. Run the `wait_for_mssql` command every second or half-second until it succeeds or we reach the max time.

Reply:
this function was copy-pasted from Hugh's `bash-lib`. didn't go into details too much.